CONTRIBUTING.md

```diff
@@ -0,0 +1,256 @@
+  exit 1
+fi
+if [ -d $SRC_DIR ] ; then
+  echo "Source dir is $SRC_DIR"
+else
+  echo  "Source dir $SRC_DIR does not exist"
+  exit 1
+fi
+ABS_SRC_DIR=`realpath $SRC_DIR`
+if [ -z $2 ] ; then
+  BUILD_DIR="build-${SRC_DIR}"
+else
+  BUILD_DIR=$2
+fi
+echo "Build dir is $BUILD_DIR"
+if [ -d $BUILD_DIR ] ; then
+ echo "Build dir exists. Cmake will re-use cached data." 
+else
+ mkdir $BUILD_DIR
+fi
+cd $BUILD_DIR
+cmake $CMAKE_FLAGS $ABS_SRC_DIR
+make -j 16
+
+```
+
+Compiling from a source requires patience. MySQL has many dependencies on different libraries. The good hint is to install build dependencies for the "upstream" version of MySQL. 
+For example, on Ubuntu ( don't forget to enable sources in _software-properties-gtk ) run following command:
+```
+apt build-dep mysql-server-core-8.0
+```
+
+Another hint is to check [ documentation on  compilation ](https://docs.percona.com/percona-server/8.0/compile-percona-server.html) .
+
+If it does not help, we recommend a try and fail approach to installing the development packages.
+
+When the source finally successfully compiles, you're ready to run tests and check the build. You have to run at least the "start and
+shutdown test" of mysqld.
+
+```
+cd build-percona-server/mysql-test
+./mtr main.1st
+``` 
+
+### Coding ...
+
+When the tests are OK, you're ready to set up your favorite IDE for the project development. The build script above will
+give you a clue what CMake parameters to add to the project in IDE.
+
+To be able to run and debug the _mysqld_ binary in your IDE, first run the following command in the "mysql-test" directory of the
+build tree:
+
+```
+./mtr --manual-debug main.1st
+```
+
+Then find the string stating with "args:". This is the command line arguments you should pass to the execution of "
+mysqld" in your IDE. This step allows to run/debug "mysqld" from the IDE.
+
+From this moment you're ready to code and debug! Happy coding!
+
+### Testing
+
+Prior to submitting your code, ensure it is thoroughly tested, both manually and with automated test suites. The
+maintenance of high-quality code is paramount to ensure the longevity and functionality of our project.
+
+Testing of MySQL is a whole big thing, so please go to
+the [ official documentation on MySQL testing ](https://dev.mysql.com/doc/dev/mysql-server/latest/PAGE_TESTING_TOOLS.html)
+first.
+
+If it is bug fix, make sure that corresponding tests from the test suite work fine. It's a good idea to patch existing
+tests to cover your specific case. For the introduction of new features, the development of a comprehensive test suite
+is strongly recommended.
+
+## Preparing the Pool Request
+
+After finishing the tasks outlined above, you must prepare your branch for a pull request.   
+Please do not forget to run ``` git clang-format ``` on your code before pushing to the GitHub repository.
+
+Then, squash all commits to one or a couple of logically consistent commits.
```

Comment:
It's confusing. I understand squashing to one, but how do you squash to a couple of commits? What makes a commit logically consistent?

Reply:
There is an example below that explains the point.